parent/main/resources/checkstyle.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" ?>
+<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
+      "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
+
+<module name="Checker">
+   <!--
+       If you set the basedir property below, then all reported file
+       names will be relative to the specified directory. See
+       http://checkstyle.sourceforge.net/5.x/config.html#Checker
+       <property name="basedir" value="${basedir}"/>
+   -->
+
+   <!-- Checks for whitespace                               -->
+   <!-- See http://checkstyle.sf.net/config_whitespace.html -->
+   <module name="FileTabCharacter"/>
+
+   <module name="FileLength"/>
+
+   <module name="FileTabCharacter"/>
+
+   <module name="TreeWalker">
+
+      <!-- Checks for Naming Conventions.                  -->
+      <!-- See http://checkstyle.sf.net/config_naming.html -->
+      <!--<module name="ClassTypeParameterName"/>-->
+      <module name="ConstantName">
+         <property name="format"
+                   value="^[log(ger)?|[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$|trace|debug]"/>
```

Comment:
put $ at the end for all expressions (this matches 'traceFoo', while we want to allow only 'trace')